### PR TITLE
re-enable vsphere integration tests

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
Now that we have a working vSphere setup, this PR undoes the changes from #8841.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/vsphere/folder_test.go [hunk 1]
```diff
@@ -64,30 +64,26 @@ func TestProvider_GetVMFolders(t *testing.T) {
 		name            string
 		dc              *kubermaticv1.DatacenterSpecVSphere
 		expectedFolders sets.String
-		// If we check for folders on the root we might have other tests creating folders, so we have to skip that check
-		skipAdditionalFoldersCheck bool
```
Comment (xrstf) [inline pkg/provider/cloud/vsphere/folder_test.go:68]: There was a folder named after a cluster in our vCenter. Assuming that can always happen, I decided to make all folder tests to only check for a subset, not equality.

File: pkg/provider/cloud/vsphere/network_test.go [hunk 2]
```diff
@@ -33,16 +33,10 @@ func TestGetPossibleVMNetworks(t *testing.T) {
...
-				},
-				{
-					AbsolutePath: "/kubermatic-e2e/network/e2e-networks/subfolder/e2e-distributed-switch-uplinks",
-					RelativePath: "e2e-networks/subfolder/e2e-distributed-switch-uplinks",
-					Type:         "DistributedVirtualPortgroup",
-					Name:         "e2e-distributed-switch-uplinks",
+					AbsolutePath: "/dc-1/network/VM Network",
+					RelativePath: "VM Network",
+					Type:         "Network",
+					Name:         "VM Network",
```
Comment (xrstf) [inline pkg/provider/cloud/vsphere/network_test.go:39]: Those values look weird, but it's what we get from the API. No idea if this is correct.

Comment (moadqassem) [reply pkg/provider/cloud/vsphere/network_test.go:39]: They are correct. Our vCenter atm doesn't use dvSwitch they use svswitch.

**Review by moadqassem — APPROVED** [review]
/approve 
/lgtm

**Pull request merged**